rewards/utils/emission_utils.py

```diff
@@ -32,6 +33,39 @@ def get_flat_emission_rate(sett: str, chain: str) -> Decimal:
     return Decimal(get_emission_control(chain).proRataEmissionRate(sett).call() / 1e4)
 
 
+@lru_cache
+def get_nft_control(chain: Network) -> ContractFunctions:
+    return make_contract(
+        EMISSIONS_CONTRACTS[chain]["NFTControl"], Abi.NFTControl, chain
+    )
+
+
+@lru_cache
+def get_nft_weights(chain: Network):
+    nft_control = get_nft_control(chain)
+    schedules = list(
+        map(
+            lambda ws: NFTWeightSchedule(ws[0], ws[1], ws[2], ws[3]),
+            nft_control.getNftWeightSchedules().call(),
+        )
+    )
+    weights = {}
+    for weight_schedule in schedules:
+        key = f"{weight_schedule.addr}-{weight_schedule.nft_id}"
+        if key not in weights:
+            weights[key] = weight_schedule
+        else:
+            curr_schedule = weights[key]
+            if weight_schedule.timestamp > curr_schedule.timestamp:
+                weights[key] = weight_schedule
+    return weights
+
+
+def get_nft_weight(chain: str, nft_address: str, nft_id: int) -> Decimal:
+    weights = get_nft_weights(chain)
+    return Decimal(weights[f"{nft_address}-{nft_id}"].weight / 1e18)
```

Comment:
2 comments here - let's assign the key in a separate variable to make it more understandable, and we should handle the case where `weights` is undefined or the key is not in the dict.